chain/client/src/stateless_validation/state_witness_distribution_actor.rs

```diff
@@ -0,0 +1,59 @@
+use actix::Actor;
+use near_async::messaging::Sender;
+use near_async::{MultiSend, MultiSendMessage, MultiSenderFrom};
+use near_network::types::PeerManagerAdapter;
+use near_o11y::{handler_debug_span, WithSpanContext};
+use near_performance_metrics_macros::perf;
+use near_primitives::stateless_validation::SignedEncodedChunkStateWitness;
+use near_primitives::types::AccountId;
+
+use super::state_witness_distribution_actions::StateWitnessDistributionActions;
+
+pub struct StateWitnessDistributionActor {
+    pub actions: StateWitnessDistributionActions,
+}
+
+impl StateWitnessDistributionActor {
+    pub fn spawn(network_adapter: PeerManagerAdapter) -> (actix::Addr<Self>, actix::ArbiterHandle) {
+        let arbiter = actix::Arbiter::new().handle();
+        let addr = Self::start_in_arbiter(&arbiter, |_ctx| Self {
+            actions: StateWitnessDistributionActions::new(network_adapter),
+        });
+        (addr, arbiter)
+    }
+}
+
+impl actix::Actor for StateWitnessDistributionActor {
+    type Context = actix::Context<Self>;
+}
+
+#[derive(actix::Message, Debug)]
+#[rtype(result = "()")]
+pub struct DistributeChunkStateWitnessRequest {
+    pub chunk_validators: Vec<AccountId>,
+    pub signed_witness: SignedEncodedChunkStateWitness,
+}
+
+#[derive(Clone, MultiSend, MultiSenderFrom, MultiSendMessage)]
+#[multi_send_message_derive(Debug)]
+pub struct StateWitnessDistributionSenderForClient {
+    pub distribute_chunk_state_witness: Sender<DistributeChunkStateWitnessRequest>,
+}
+
+impl actix::Handler<WithSpanContext<DistributeChunkStateWitnessRequest>>
+    for StateWitnessDistributionActor
+{
+    type Result = ();
+
+    #[perf]
+    fn handle(
+        &mut self,
+        msg: WithSpanContext<DistributeChunkStateWitnessRequest>,
+        _: &mut Self::Context,
+    ) -> Self::Result {
+        let (_span, msg) = handler_debug_span!(target: "client", msg);
+        if let Err(err) = self.actions.handle_distribute_chunk_state_witness_request(msg) {
```

Comment:
not sure about the naming conversion but can this be just "distribute_chunk_state_witness"?

Reply:
Umm, for consistency, I'm just keeping all calls from actor to actions as `handle_xyz_request`